sdk/src/main/java/com/mapbox/maps/viewannotation/ViewAnnotationManagerImpl.kt

```diff
@@ -0,0 +1,250 @@
+package com.mapbox.maps.viewannotation
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewTreeObserver
+import android.widget.FrameLayout
+import androidx.annotation.LayoutRes
+import androidx.asynclayoutinflater.view.AsyncLayoutInflater
+import com.mapbox.maps.*
+import com.mapbox.maps.plugin.ViewPlugin
+import java.util.concurrent.ConcurrentHashMap
+
+internal class ViewAnnotationManagerImpl(
+  private val mapView: MapView,
+  private val mapboxMap: MapboxMap = mapView.getMapboxMap()
+) : ViewAnnotationManager, ViewAnnotationPositionsListener {
+
+  init {
+    mapView.requestDisallowInterceptTouchEvent(false)
+    mapboxMap.setViewAnnotationPositionsUpdateListener(this)
+  }
+
+  private val asyncInflater by lazy {
+    try {
+      AsyncLayoutInflater(mapView.context)
+    } catch (e: NoClassDefFoundError) {
+      throw RuntimeException(
+        "Please add https://mvnrepository.com/artifact/androidx.asynclayoutinflater/asynclayoutinflater/1.0.0 dependency " +
+          "to your project to make use of asynchronous view inflation when adding view annotation!"
+      )
+    }
+  }
+
+  private val annotationMap = ConcurrentHashMap<String, ViewAnnotation>()
+  private val idLookupMap = HashMap<View, String>()
```

Comment:
Why is it not ConcurrentHashMap as well?

Reply:
Further question - is this separate map needed at all? It just helps to look up the id in add/remove/updateAnnotation, does it? If so - it seems it's possible to get it from the `annitationMap.entries` list.

Reply:
it's for optimization needs: `idLookupMap` is needed cause user operates with `View` while all the internal logic operates with `id: String`. So in order not to use N complexity each time going thru all `annotationMap` entries and checking if `view == entry.value.view` I introduced lookup map.

Reply:
Yeah, but `idLookupMap` map is used only by user code in add/remove/update calls, right? Since it's called very occasionally and not at 60 fps, plus number of platform views won't be that big (cant imagine more than couple dozens of them worst case) IMO it's not worth it to complicate to code here

Reply:
we can never decide for the user - at least for update operation it's quite easy to imagine that it could also be called 60 times per sec e.g. to dynamically resize view annotation width and height based on current zoom level.